presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/InstallationManager.java

```diff
@@ -17,78 +17,147 @@
 import com.google.gson.Gson;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
-import com.google.inject.Inject;
 import com.teradata.prestomanager.common.StopType;
 import io.airlift.log.Logger;
+import org.glassfish.jersey.client.JerseyClientBuilder;
 
 import javax.ws.rs.ProcessingException;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriBuilder;
 
 import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Map;
 
-import static com.teradata.prestomanager.agent.AgentFileUtils.getFileProperty;
-import static com.teradata.prestomanager.agent.CommandExecutor.execCommandResult;
-import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.agent.PrestoUtils.getPrestoPort;
+import static com.teradata.prestomanager.agent.PrestoUtils.isRunningCoordinator;
+import static com.teradata.prestomanager.common.StopType.TERMINATE;
 import static java.lang.String.format;
 import static javax.ws.rs.client.Entity.entity;
 import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
 import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
 import static javax.ws.rs.core.Response.Status.ACCEPTED;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
 import static javax.ws.rs.core.Response.Status.CONFLICT;
 import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 import static javax.ws.rs.core.Response.Status.OK;
 import static javax.ws.rs.core.UriBuilder.fromUri;
 
-public class PrestoRpmController
+public abstract class InstallationManager
```

Comment:
I would split this class into an abstract class and one or two interfaces.

This class would still contain the methods that install, uninstall, and upgrade presto, and `getVersion`, but the other functionality it has would be delegated to the other class.

The interface (maybe `PrestoController`) would have the methods that start and stop Presto, and `isRunning`.

`RpmInstallationManager` could implement the interface, so it would not need any changes.

The only issue that arises is that the start/stop classes would not have access to `isInstalled`. To solve this, I would have the caller of the functions perform the check, which also helps by eliminating the repeated `if (!isInstalled) return NOT_FOUND` blocks in each of the start/stop functions.